packages/blockchain-wallet-v4-frontend/src/data/components/buySell/sagas.ts

```diff
@@ -812,6 +824,47 @@ export default ({ api, coreSagas, networks }: { api: APIType; coreSagas: any; ne
     }
   }
 
+  const fetchBuyQuote = function* ({ payload }: ReturnType<typeof A.fetchBuyQuote>) {
+    while (true) {
+      try {
+        yield put(A.fetchBuyQuoteLoading())
+
+        const { amount, pair, paymentMethod } = payload
+        const pairReversed = reversePair(pair)
+        let paymentMethodId: string | undefined
+
+        if (paymentMethod === BSPaymentTypes.BANK_TRANSFER) {
+          paymentMethodId = 'd74992e0-a462-4aeb-903e-fc8e9a11bb40'
+        }
+
+        const quote: ReturnType<typeof api.getBuyQuote> = yield call(
+          api.getBuyQuote,
+          pairReversed,
+          'SIMPLEBUY',
+          amount,
+          paymentMethod,
+          paymentMethodId
+        )
+
+        yield put(
+          A.fetchBuyQuoteSuccess({
+            fee: quote.feeDetails.fee.toString(),
+            pair,
+            quote,
+            rate: parseInt(quote.price)
+          })
+        )
+        const refresh = -moment().diff(quote.quoteExpiresAt)
```

Comment:
I didn't understand this line, why is it negative?

Reply:
@plondon may be able to explain this one. Copied from `fetchSellQuote()`

Reply:
Hmm, I think I get it, as the diff from the `quoteExpiresAt ` will be negative, he double negates it. Smart move.